src/compas_timber/elements/fastener.py

```diff
@@ -185,32 +207,34 @@ def shape(self):
             if self.shapes:
                 geometries.extend(self.shapes)
 
-            self._shape = geometries[0]
-            for geometry in geometries[1:]:
-                self._shape += geometry
+            self._shape = None
+            if geometries:
+                self._shape = geometries[0]
+                for geometry in geometries[1:]:
+                    self._shape += geometry
         return self._shape
 
     @property
     def geometry(self):
         """returns the geometry of the interface in the model (oriented on the timber element)"""
-        return self.shape.transformed(Transformation.from_frame(self.frame))
+        return self.shape.transformed(Transformation.from_frame(self.frame)) if self.shape else None
```

Comment:
`geometry` is already implemented in `Element` and it calls `compute_geometry` (which you implemented in the parent class above)

Reply:
this is not an `Element`